recipes/jsoncpp/all/conanfile.py

```diff
@@ -87,34 +93,30 @@ def package(self):
             }
         )
 
-    @staticmethod
-    def _create_cmake_module_alias_targets(module_file, targets):
+    def _create_cmake_module_alias_targets(self, module_file, targets):
         content = ""
         for alias, aliased in targets.items():
-            content += textwrap.dedent("""\
+            content += textwrap.dedent(f"""\
                 if(TARGET {aliased} AND NOT TARGET {alias})
                     add_library({alias} INTERFACE IMPORTED)
                     set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
                 endif()
-            """.format(alias=alias, aliased=aliased))
-        tools.save(module_file, content)
-
-    @property
-    def _module_subfolder(self):
-        return os.path.join("lib", "cmake")
+            """)
+        save(self, module_file, content)
 
     @property
     def _module_file_rel_path(self):
-        return os.path.join(self._module_subfolder,
-                            "conan-official-{}-targets.cmake".format(self.name))
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "jsoncpp"
-        self.cpp_info.names["cmake_find_package_multi"] = "jsoncpp"
-        self.cpp_info.names["pkg_config"] = "jsoncpp"
-        self.cpp_info.builddirs.append(self._module_subfolder)
-        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
-        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.set_property("cmake_file_name", "jsoncpp")
+        self.cpp_info.set_property("cmake_target_name", "jsoncpp_lib")
+        if not self.options.shared:
+            self.cpp_info.set_property("cmake_target_aliases", ["jsoncpp_static", "jsoncpp_lib_static"])
+        self.cpp_info.set_property("pkg_config_name", "jsoncpp")
+        self.cpp_info.libs = collect_libs(self)
```

Comment:
```suggestion
        self.cpp_info.libs = ["jsoncpp"]
```
https://github.com/open-source-parsers/jsoncpp/blob/1.9.5/src/lib_json/CMakeLists.txt#L119

I just checked current Conan packages available in Conan Center and is always `libjsoncpp` or `jsoncpp.lib` or `jsoncpp.dll`

Reply:
Nop: https://github.com/open-source-parsers/jsoncpp/blob/1.9.5/src/lib_json/CMakeLists.txt#L142-L157

Reply:
So we are missing something:

```
% conan install -r conancenter jsoncpp/1.9.5@ -g cmake_find_package -pr:h=msvc -pr:b=msvc -o jsoncpp:shared=True
Configuration (profile_host):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=None
compiler.runtime=MD
compiler.toolset=None
compiler.version=16
os=Windows
os_build=Windows
[options]
jsoncpp:shared=True
[build_requires]
[env]
[conf]

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=None
compiler.runtime=MD
compiler.toolset=None
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
[conf]

jsoncpp/1.9.5: Retrieving from server 'conancenter' 
jsoncpp/1.9.5: Trying with 'conancenter'...
Downloading 8e01d356711afae4c555375dd7061080ddb0b34bb5f8806ddc26341e7d2cdd6f completed [0.65k]                         
Downloading 3a294815ac3d07f64bfcdfaed079efee545c06538e22c973024e4f49a6646d76 completed [4.90k]                           
Downloading 5062b68cfbdd991864a834596f3788797c778ba9c7f6e4e3ddc4465a8e413655 completed [0.25k]                         
Decompressing conan_export.tgz completed [0.00k]                                         
jsoncpp/1.9.5: Downloaded recipe revision 536d080aa154e5853332339bf576747c
Installing package: jsoncpp/1.9.5
Requirements
    jsoncpp/1.9.5 from 'conancenter' - Downloaded
Packages
    jsoncpp/1.9.5:127af201a4cdf8111e2e08540525c245c9b3b99e - Download

Installing (downloading, building) binaries...
jsoncpp/1.9.5: Retrieving package 127af201a4cdf8111e2e08540525c245c9b3b99e from remote 'conancenter' 
Decompressing conan_package.tgz completed [0.00k]                                        
jsoncpp/1.9.5: Package installed 127af201a4cdf8111e2e08540525c245c9b3b99e
jsoncpp/1.9.5: Downloaded package revision 7913b82997164f4adf246ecc162d954e
Generator cmake_find_package created Findjsoncpp.cmake
Aggregating env generators

% ls ~/.conan/data/jsoncpp/1.9.5/_/_/package/127af201a4cdf8111e2e08540525c245c9b3b99e/lib                       
cmake		jsoncpp.lib 
```

Reply:
Ok, there is a suffix in static lib only if shared is built as well.